Userland/Libraries/LibWeb/DOM/Element.h

```diff
@@ -91,6 +93,10 @@ class Element
     const ShadowRoot* shadow_root() const { return m_shadow_root; }
     void set_shadow_root(RefPtr<ShadowRoot>);
 
+    Optional<CSS::StyleResolver::CustomPropertyResolutionTuple> resolve_custom_property(const String& custom_property_name) { return m_custom_properties.get(custom_property_name); }
+    bool has_custom_property(const String& custom_property_name) { return m_custom_properties.get(custom_property_name).has_value(); }
+    void add_custom_property(const String& custom_property_name, CSS::StyleResolver::CustomPropertyResolutionTuple style_property) { m_custom_properties.set(custom_property_name, style_property); }
```

Comment:
These lines are p r e t t y  d a m n  l o n g

Reply:
I just broke them up, still looong, but wayy more manageable now.

Reply:
loooong is definitely better than p r e t t y d a m n l o n g : )